nav: close dot prompt when help is opened

JonasBa opened the pull request:
Do not show any indicators if user has opted into chonk already and wire up dismiss actions with help menu opening

JonasBa committed:
nav: close dot prompt when help is opened

JonasBa committed:
chonk: do not show prompt if user has opted into chonk

JonasBa committed:
chonk: dismiss indicator and tooltip prompts

JonasBa commented on static/app/components/sidebar/help.tsx:
```diff
@@ -102,7 +111,7 @@ function SidebarHelp({orientation, collapsed, hidePanel, organization}: Props) {
                     );
                   }}
                 >
-                  {t('Try New Navigation')} <Badge type="alpha">Alpha</Badge>
```
This was just a missing translation that I added

TkDodo commented on static/app/components/sidebar/help.tsx:
```diff
@@ -29,15 +31,21 @@ type Props = Pick<
   const {mutate: mutateUserOptions} = useMutateUserOptions();
   const openForm = useFeedbackForm();
 
+  const onHelpMenuOpen = useCallback(() => {
+    navPrompts.onOpenHelpMenu();
+    chonkPrompts.dismissDotIndicatorPrompt();
+  }, [navPrompts, chonkPrompts]);
```
there’s no need to memoize this with `useCallback`. For one, `navPrompts.onOpenHelpMenu` is an inline function upon creation:

https://github.com/getsentry/sentry/blob/d5f5d49ec068ac8503e71508596736580c901ef2/static/app/views/nav/useNavPrompts.tsx#L42-L46

and also, we’re only using the resulting function to pass it to `onOpen` of `DeprecatedDropdownMenu` which is neither a pure component nor does it use the function in memoization or effects (which it shouldn’t do anyway).

That means the memoization will be re-calculated on every render anyways, and we don’t need it. I’m wondering where the instinct to wrap things in `useMemo` and `useCallback` in the sentry code-base comes from 🤔. This doesn’t make the code any faster, if anything, it’ll make it “slower” because there will be one additional caching react has to perform. It also makes the codebase a lot harder to read if this is done in every component.

```suggestion
  const onHelpMenuOpen = () => {
    navPrompts.onOpenHelpMenu();
    chonkPrompts.dismissDotIndicatorPrompt();
  };
```

JonasBa replied:
The instinct here is self-preservational. The codebase is full of effects under untested paths, and I've had changes like these blow up on me so often that I err on the safe side of things. That said though, this is unnecessary in this situation indeed, and should be safe to remove

TkDodo replied:
understandable, we need to work on the root cause :)

TkDodo commented on static/app/views/nav/primary/help.tsx:
```diff
@@ -55,10 +56,12 @@ export function PrimaryNavigationHelp() {
   return (
     <StackedNavigationTourReminder>
       <SidebarMenu
+        onClick={chonkPrompt.dismissDotIndicatorPrompt}
```
I think we should call this `onItemClick?` or something, as it gets triggered when an item is clicked, not when the `SideBarMenu` is clicked

JonasBa replied:
Is it `onAction` that you are referring to here? The different namings of these callbacks are such PITA, because they differ between each component (I could have sworn I looked over the type to see if there is anything)

JonasBa replied:
Agh, actually onAction is just for individual actions. I don't think this component supports anything like what you are referring to actually :/

JonasBa committed:
chonk: remove callback

JonasBa committed:
chonk: dismiss indicator and tooltip prompts (#89977)

Add Chonk UI Banner prompt to legacy and new nav

![CleanShot 2025-04-21 at 11 50
14@2x](https://github.com/user-attachments/assets/db10d143-5bec-4302-8b65-70ce4108a8d0)

![CleanShot 2025-04-21 at 12 10
19@2x](https://github.com/user-attachments/assets/139900b6-ec6b-4c12-bfb6-326b4e2de346)

JonasBa committed:
rename to onOpen
